src/turtle.rs

```diff
@@ -199,6 +199,46 @@ impl Turtle {
         block_on(self.turtle.wait(secs))
     }
 
+    /// Instructs the turtle to draw an arc by the given angle for a given radius in the given step count from the current position of the turtle.
+    ///
+    /// The `radius` parameter is a floation point number that represents the radius of the arc
+    /// that you want to draw. If a negative `radius` is passed its absolute value is
+    /// considered for drawing the arc.
+    ///
+    /// The `extent` parameter is a floating point number that represents how much you want the
+    /// turtle to rotate. If the `extent` parameter is `None` then it is defaulted to 360. If
+    /// a negative `extent` is passed the arc is drawn in the opposite direction.
+    ///
+    /// the `steps` parameter is an integer that represent over how many steps you want the turtle to draw. If the `steps` parameter
+    /// is `None` then it is defaulted to the arc_length / 4.0.
+    ///
+    /// #Example
+    ///
+    /// ```rust
+    /// # use turtle::*;
+    /// # let mut turtle = Turtle::new();
+    /// // Turtle will draw an arc of 90 in a clockwise direction
+    /// turtle.arc(100.0, Some(90.0), None);
+    ///
+    /// // Turtle will draw an arc of 90 in a anti-clockwise direction
+    /// turtle.arc(-100.0, Some(90.0), None);
+    ///
+    /// // Turtle will draw an arc of 90 in a clockwise direction
+    /// turtle.arc(-100.0, Some(-90.0), None);
+    ///
+    /// //Turtle will draw an arc of 90 in a anti-clockwise direction
+    /// turtle.arc(-100.0, Some(90.0), None);
+    ///
+    /// // Turtle will draw an arc of 90 in a clockwise direction in 10 steps
+    /// turtle.arc(100.0, Some(90.0), Some(10));
+    ///
+    /// // Turtle will draw an arc of 90 in a clockwise direction in 100 steps
+    /// turtle.arc(100.0, Some(90.0), Some(100));
```

Comment:
```suggestion
    /// // Both of these lines will cause the turtle to draw
    /// // a 90 degree clockwise arc (by turning right)
    /// turtle.arc(100.0, Some(90.0), None);
    /// turtle.arc(-100.0, Some(-90.0), None);
    ///
    /// // Both of these lines will cause the turtle to draw
    /// // a 90 degree counterclockwise arc (by turning left)
    /// turtle.arc(-100.0, Some(90.0), None);
    /// turtle.arc(100.0, Some(-90.0), None);
    ///
    /// // A 90 degree clockwise arc in 10 steps
    /// turtle.arc(100.0, Some(90.0), Some(10));
    /// // A 90 degree clockwise arc in 100 steps
    /// turtle.arc(100.0, Some(90.0), Some(100));
```

When using an example to try to teach the subtleties of an API to someone, it's good to group related statements together so readers can see which small differences lead to the same outcome. I've changed this example to group some of the lines you added together so the user can see how the signs of the radius and extent affect the output.

Note: If your other documentation about us taking the absolute value of the radius is true, are these examples even accurate? I think this should be fine once we fix the code to no longer take the absolute value. We should definitely make sure we're accurate about the behaviour of our code in our examples!